osmaxx-py/osmaxx/contrib/auth/frontend_permissions.py

```diff
@@ -45,3 +46,26 @@ class FrontendAccessRequiredMixin(object):
     @method_decorator(frontend_access_required)
     def dispatch(self, *args, **kwargs):
         return super(FrontendAccessRequiredMixin, self).dispatch(*args, **kwargs)
+
+
+class AuthenticatedAndAccessPermission(permissions.BasePermission):
+    """
+    Allows access only to authenticated users with frontend permissions.
+    """
+
+    def has_permission(self, request, view):
+        return request.user.is_authenticated() and _may_user_access_osmaxx_frontend(request.user)
+
+
+class HasBBoxAccessPermission(permissions.BasePermission):
+    message = 'Accessing this bounding box is not allowed.'
+
+    def has_object_permission(self, request, view, obj):
+        return obj.excerpt.is_public or obj.excerpt.owner == request.user
```

Comment:
Yes, that would make sense. It would actually make sense to not have names like `orderer`, but actually always use `owner` or alternatively `user`, though I prefer the former. An abstract base class would most likely make sense here, which has both: an owner field and the _duck-interface_.